Support for Rails Credentials in addition to Env Vars

benhayehudi opened the pull request:
This PR introduces the ability to instantiate a Nexmo client in a Rails app by either credentials or environment variables. It conditions the Nexmo client config values based on the presence of  `Rails.application.credentials.nexmo`.

fabianrbz commented on CHANGES.md:
```diff
@@ -1,3 +1,7 @@
\ No newline at end of file
+* First version!
+
+# 0.2.0
```
Nitpick: Usually the newest changes are at the top, so you don't need to scroll all the way down in order to see the latest changes.

fabianrbz commented on nexmo_rails.gemspec:
```diff
@@ -21,6 +21,7 @@ Gem::Specification.new do |spec|
   spec.add_dependency('dotenv-rails')
   spec.add_development_dependency('rspec')
   spec.add_development_dependency('generator_spec')
+  spec.add_development_dependency('rails-dummy')
```
@benhayehudi do we need this dependency?

fabianrbz commented on lib/generators/nexmo_initializer/nexmo_initializer_generator.rb:
```diff
@@ -1,11 +1,34 @@
+
+    def credentials_initializer
+      initializer "nexmo.rb" do <<~HEREDOC
+        Nexmo.setup do |config|
```
nitpick: I prefer to have the `=` aligned :p, but you can leave it as it is

benhayehudi commented:
> @benhayehudi i left some comments, mostly nitpicks.
> It looks good 💪 .
> 
> The only thing is that CI is not set up for this gem so I had to manually run the specs to be sure they were passing. I would say that we can tackle that as part of a different issue, but it's definitely something that we should address. WDYT?

Thank for the lookover! I addressed the issues you raised. I left the `=` sign as is since it might bug me to keep on making sure they were all aligned all the time. :) 

Setting up the CI for the gem I think is a good next step to implement to enable better processes for future work on it.

benhayehudi commented:
Thanks @fabianrbz -- I'm going to hold off on merging for a little bit, so going to add a `don't merge` label for now. Appreciate the review!
